gpii/node_modules/flowManager/configs/gpii.flowManager.config.production.json

```diff
@@ -7,7 +7,7 @@
                 "record": {
                     "type": "kettle.dataSource.URL",
                     "options": {
-                        "url": "https://preferences.gpii.net/preferences/%userToken"
+                        "url": "@expand:kettle.resolvers.env(GPII_FLOWMANAGER_PREFERENCES_URL, https://preferences.gpii.net/preferences/%userToken)"
```

Comment:
Ok, I see now that you have both defaulted and configured the URL in two places. To avoid making a mess, these definitions should just appear in one place. It looks like you want this to be the standard configuration for all production configurations of the flowManager, in which case this is the right place to do this here. This can be done without the defaulting mechanism for the resolver by rewriting the options distribution block as follows:

```
        "distributeOptions": {
            "preferencesDataSource.productionDataSource": {
                "record": {
                    "type": "kettle.dataSource.URL",
                    "options": {
                        "url": "https://preferences.gpii.net/preferences/%userToken"
                    }
                 },
                 "target": "{that flowManager preferencesDataSource}"
             },
             "preferencesDataSource.envUrl": {
                 "record": "@expand:kettle.resolvers.env(GPII_FLOWMANAGER_PREFERENCES_URL)",
                 "target": "{that flowManager.preferencesDataSource}.options.url",
                 "priority": "after:preferencesDataSource.productionDataSource"
             }
        }
```